specs/Workers.md

```diff
@@ -0,0 +1,1399 @@
+  HRESULT Invoke([in] HRESULT errorCode, [in] ICoreWebView2ServiceWorkerRegistrationCollectionView* result);
+}
+
+/// Receives the result of the `GetServiceWorkerRegistration` method.
+[uuid(e5b37473-5833-5ec1-b57e-ccbbbc099a32), object, pointer_default(unique)]
+interface ICoreWebView2GetServiceWorkerRegistrationCompletedHandler : IUnknown {
+
+  /// Provides the result of the corresponding asynchronous method.
+  HRESULT Invoke([in] HRESULT errorCode, [in] ICoreWebView2ServiceWorkerRegistration* result);
+}
+
+/// Receives `ServiceWorkerRegistered` events.
+[uuid(81262ddf-ef6c-5184-8ae0-01fa0a4c86c7), object, pointer_default(unique)]
+interface ICoreWebView2ServiceWorkerRegisteredEventHandler : IUnknown {
+  /// Provides the event args for the corresponding event.
+  HRESULT Invoke(
+      [in] ICoreWebView2ServiceWorkerManager* sender,
+      [in] ICoreWebView2ServiceWorkerRegisteredEventArgs* args);
+}
+
+/// Event args for the `ServiceWorkerRegistered` event.
+[uuid(a18bd62a-8246-5d2b-abc5-2bdbbc13767b), object, pointer_default(unique)]
+interface ICoreWebView2ServiceWorkerRegisteredEventArgs : IUnknown {
+  /// The service worker that was registered.
+  HRESULT GetServiceWorkerRegistration(
+      [out, retval] ICoreWebView2ServiceWorkerRegistration** value
+  );
+}
+
+/// A collection of ICoreWebView2ServiceWorkerRegistration.
+[uuid(9860773f-16e5-5521-9bc2-a6d8e5347d4a), object, pointer_default(unique)]
+interface ICoreWebView2ServiceWorkerRegistrationCollectionView : IUnknown {
+  /// The number of elements contained in the collection.
+  [propget] HRESULT Count([out, retval] UINT32* value);
+
+  /// Gets the element at the given index.
+  HRESULT GetValueAtIndex([in] UINT32 index, [out, retval] ICoreWebView2ServiceWorkerRegistration** value);
+}
+
+/// Receives the result of the `GetServiceWorker` method.
+[uuid(bd120b47-b859-5f44-9913-0318130ffb69), object, pointer_default(unique)]
+interface ICoreWebView2GetServiceWorkerCompletedHandler : IUnknown {
+
+  /// Provides the result of the corresponding asynchronous method.
+  HRESULT Invoke([in] HRESULT errorCode, [in] ICoreWebView2ServiceWorker* result);
+}
+
+/// Receives `Destroyed` events.
+[uuid(9d79117e-4e96-5a53-9933-d7a31d0106ae), object, pointer_default(unique)]
+interface ICoreWebView2ServiceWorkerRegistrationDestroyedEventHandler : IUnknown {
+  /// Provides the event args for the corresponding event.
+  HRESULT Invoke(
+      [in] ICoreWebView2ServiceWorkerRegistration* sender,
+      [in] IUnknown* args);
+}
+
+[uuid(0c0b03bd-ced2-5518-851a-3d3f71fb5c4b), object, pointer_default(unique)]
+interface ICoreWebView2ServiceWorkerRegistration : IUnknown {
+  /// The `scope` parameter is a fully qualified URL that defines the range of URLs
+  /// a service worker can control.
+  ///
+  /// By default, the `scope` is the directory where the service worker script resides.
+  /// For example, if the script is at https://example.com/sw.js, the default scope is
```

Comment:
It would be good to use https://example.com/app/sw.js as your example. You say the default is the directory where the service worker script resides but the example may lead people to think its just the origin with root path. I would guess for https://example.com/app/sw.js the default scope is https://example.com/app/?